ITEvents/Services/UserSettingsService.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import CoreData
+import UIKit
+
+class UserSettingsService: IUserSettingsService {
+    let entityName = "Settings"
+    
+    func fetchSettings(then completion: @escaping (UserSettings) -> Void) {
+        let settings = getOrCreateSettings()
+        var userSettings = UserSettings(isListLayoutSelected: true)
+        let isList = settings.value(forKey: "isListLayoutSelected") as! Bool
+        userSettings.isListLayoutSelected = isList
+        completion(userSettings)
+    }
+    
+    func save(settings: UserSettings) {
+        let savedSettings = getOrCreateSettings()
+        savedSettings.isListLayoutSelected = settings.isListLayoutSelected
+        try? getContext().save()
+    }
+    
+    private func getContext() -> NSManagedObjectContext {
```

Comment:
I'd rather have the following method:
```swift
private func getApp() -> AppDelegate {
    return UIApplication.shared.delegate as! AppDelegate
}
```
the we can do 
```swift
getApp.saveContext()
```
instead of 
```swift
try? getContext().save()
```

Reply:
and also we'd need to updated getOrCreateSettings:
```swift
private func getOrCreateSettings() -> Settings {
       let managedContext = getApp().persistentContainer.viewContext
       ...
}
```